packages/babel-types/src/converters/valueToNode.js

```diff
@@ -16,6 +15,18 @@ import {
   binaryExpression,
 } from "../builders/generated";
 
+let nodeUtilTypes;
+try {
+  const util = require("util");
+  nodeUtilTypes = util.types || util;
+} catch {}
+
+const isRegExp = value => {
+  return nodeUtilTypes
+    ? nodeUtilTypes.isRegExp(value)
+    : Object.prototype.toString.call(value) === "[object RegExp]";
+};
```

Comment:
The `nodeUtilTypes` check only needs to be performed once:

```suggestion
const isRegExp = nodeUtilTypes?.isRegExp || value => {
  return Object.prototype.toString.call(value) === "[object RegExp]";
};
```

Reply:
Accepted, thanks!  (locally rather than via the GitHub UI, because I figured a merge from `main` would be worthwhile at the same time)

Reply:
Hrm.. looks like a small issue with the optional chaining operator.  Reading up a little on the syntax and compatibility of it now.. hopefully a small fix.

Reply:
As best I can tell, the problem was that the optional chaining operator (ES2020) wasn't available at compile-time (Node 6.x), so it's now replaced with a truthiness check.